libp2p/Host.cpp

```diff
@@ -783,15 +803,21 @@ void Host::run(boost::system::error_code const& _ec)
     unsigned reqConn = 0;
     {
         RecursiveGuard l(x_sessions);
-        for (auto const& p : m_peers)
         {
-            bool haveSession = havePeerSession(p.second->id);
-            bool required = p.second->peerType == PeerType::Required;
-            if (haveSession && required)
-                reqConn++;
-            else if (!haveSession && p.second->shouldReconnect() &&
-                        (!m_netConfig.pin || required))
-                toConnect.push_back(p.second);
+            for (auto p = m_peers.cbegin(); p != m_peers.cend(); p++)
+            {
+                bool haveSession = havePeerSession(p->second->id);
+                bool required = p->second->peerType == PeerType::Required;
+                if (haveSession && required)
+                    reqConn++;
+                else if (!haveSession)
+                {
+                    if (p->second->fallbackSeconds() == numeric_limits<unsigned>::max())
```

Comment:
Instead of using magic value of `fallbackSeconds()`, better add another method to `Peer` returning true if it had critical connection error

Reply:
@gumb0 : Good idea - I went with the magic number approach since I figured I'd have to return a large value for these cases in fallbackSeconds which would effectively prevent us from ever reconnecting, so I figured I didn't need the function. However, you're right that the code is more clear and magic numbers aren't good 😄 I'll add the function and also return value in fallbackSeconds which is large enough to prevent us from ever reconnecting but is small enough to prevent an overflow (I'm thinking 1 year in seconds)